app/lib/common/widgets/attachments/post_attachment_selection.dart

```diff
@@ -0,0 +1,185 @@
+        ref.read(attachmentDraftsProvider(widget.manager!).notifier);
+    for (var file in widget.files) {
+      await attachmentDraftsNotifier.addDraft(file);
+    }
+    await attachmentDraftsNotifier.sendDrafts();
+  }
+
+  // if room attachment, send via timeline stream message
+  void handleChatAttachmentSend(
+    List<File> selectedAttachments,
+  ) async {
+    final roomId = widget.convo!.getRoomIdStr();
+    final client = ref.read(alwaysClientProvider);
+    final inputState = ref.read(chatInputProvider(roomId));
+    final stream = widget.convo!.timelineStream();
+
+    try {
+      for (var file in selectedAttachments) {
+        final mimeType = lookupMimeType(file.path)!;
+        if (mimeType.startsWith('image/')) {
+          final bytes = file.readAsBytesSync();
+          final image = await decodeImageFromList(bytes);
+          final imageDraft = client
+              .imageDraft(file.path, mimeType)
+              .size(file.lengthSync())
+              .width(image.width)
+              .height(image.height);
+          if (inputState.repliedToMessage != null) {
+            await stream.replyMessage(
+              inputState.repliedToMessage!.id,
+              imageDraft,
+            );
+          } else {
+            await stream.sendMessage(imageDraft);
+          }
+        } else if (mimeType.startsWith('audio/')) {
+          final audioDraft =
+              client.audioDraft(file.path, mimeType).size(file.lengthSync());
+          if (inputState.repliedToMessage != null) {
+            await stream.replyMessage(
+              inputState.repliedToMessage!.id,
+              audioDraft,
+            );
+          } else {
+            await stream.sendMessage(audioDraft);
+          }
+        } else if (mimeType.startsWith('video/')) {
+          final videoDraft =
+              client.videoDraft(file.path, mimeType).size(file.lengthSync());
+          if (inputState.repliedToMessage != null) {
+            await stream.replyMessage(
+              inputState.repliedToMessage!.id,
+              videoDraft,
+            );
+          } else {
+            await stream.sendMessage(videoDraft);
+          }
+        } else {
+          final draft =
+              client.fileDraft(file.path, mimeType).size(file.lengthSync());
+          if (inputState.repliedToMessage != null) {
+            await stream.replyMessage(inputState.repliedToMessage!.id, draft);
+          } else {
+            await stream.sendMessage(draft);
+          }
+        }
+      }
+    } catch (e, s) {
+      _log.severe('error occurred sending attachments', e, s);
+    }
+
+    if (inputState.repliedToMessage != null) {
+      final notifier = ref.read(chatInputProvider(roomId).notifier);
+      notifier.setRepliedToMessage(null);
+      notifier.setEditMessage(null);
+      notifier.showReplyView(false);
+      notifier.showEditView(false);
+      notifier.setReplyWidget(null);
+      notifier.setEditWidget(null);
```

Comment:
reads like we should have a `.clear()` on the notifier, rather than have it switch the state 7 times, we could just do that once ...

Reply:
Good point, but this change isn't related to this PR, this has just been moved and was same as before!